syntax = "proto3";

package post;

service PostService {
  rpc CreatePost (CreatePostRequest) returns (PostResponse);
  rpc DeletePost (PostIdRequest) returns (Empty);
  rpc UpdatePost (UpdatePostRequest) returns (PostResponse);
  rpc GetPostById (PostIdRequest) returns (PostResponse);
  rpc ListPosts (ListPostsRequest) returns (ListPostsResponse);
}

message Post {
  int32 id = 1;
  string title = 2;
  string description = 3;
  int32 creator_id = 4;
  string created_at = 5;
  string updated_at = 6;
  bool is_private = 7;
  repeated string tags = 8;
}

message CreatePostRequest {
  string title = 1;
  string description = 2;
  int32 creator_id = 3;
  bool is_private = 4;
  repeated string tags = 5;
}

message UpdatePostRequest {
  int32 id = 1;
  string title = 2;
  string description = 3;
  bool is_private = 4;
  repeated string tags = 5;
}

message PostIdRequest {
  int32 id = 1;
}

message ListPostsRequest {
  int32 page = 1;
  int32 page_size = 2;
}

message PostResponse {
  Post post = 1;
}

message ListPostsResponse {
  repeated Post posts = 1;
}

message Empty {}
